experimental/incremental/executor_test.go

```diff
@@ -224,3 +225,76 @@ func TestCyclic(t *testing.T) {
 		result[0].Fatal.Error(),
 	)
 }
+
+func TestUnchanged(t *testing.T) {
+	t.Parallel()
+	assert := assert.New(t)
+
+	ctx := context.Background()
+	exec := incremental.New(
+		incremental.WithParallelism(4),
+	)
+
+	queries := make([]incremental.Query[int], 16)
+	for i := range queries {
+		queries[i] = ParseInt{"42"}
+	}
+
+	// Hammer the same query many, many times to ensure consistency across
+	// parallelized and serialized calls.
+	const (
+		runs     = 4
+		gsPerRun = 16
+	)
+
+	var wg, barrier sync.WaitGroup
+	for i := 0; i < runs; i++ {
+		exec.Evict(ParseInt{"42"})
+		results, _ := incremental.Run(ctx, exec, queries...)
+		for j, r := range results[1:] {
```

Comment:
Why is this ignoring the first result?

Reply:
It's not, because it's used in the body of every loop iteration, but I'll just loop over the whole slice bc it makes no difference.